src/graphorm/index.ts

```diff
@@ -171,17 +174,40 @@ const obj = new GraphORM({
           isMany: false,
           customRelation: (ctx, parentAlias, childAlias, qb): QueryBuilder =>
             qb
-              .select('*')
-              .from(SourceMember, 'sm')
-              .where(`sm."userId" = :userId`, { userId: ctx.userId })
-              .andWhere(`sm."sourceId" = "${parentAlias}".id`),
+              .addSelect('"userId"')
+              .where(`${childAlias}."userId" = :userId`, { userId: ctx.userId })
+              .andWhere(`${childAlias}."sourceId" = "${parentAlias}".id`),
         },
-        transform: nullIfNotLoggedIn,
       },
     },
   },
   SourceMember: {
     requiredColumns: ['createdAt'],
+    fields: {
+      permissions: {
+        transform: (_, ctx: Context, member: SourceMember) => {
+          if (!ctx.userId || member.userId !== ctx.userId) {
+            return null;
+          }
+
+          return (
+            roleSourcePermissions[member.role] ?? roleSourcePermissions.member
+          );
+        },
+      },
```

Comment:
Do we need this one as a query?
It seems we can determine it with the roleRank already right?

Reply:
With role rank, we can only determine the level of access privilege, but whether you have permission to actually do certain actions, it is based on this array.